daml/ContingentClaims/Claim.daml

```diff
@@ -114,29 +114,30 @@ isBearerExecutor = cata \case
 
 -- | Converts a `Serialized.Claim` read from ledger, to a more general `Claim`, for use with `Daml.Control.Recursion`.
 deserialize : Serialized.Claim t a -> Claim Observation t a
-deserialize = ana \case
+deserialize = futu \case
   Serialized.Zero -> ZeroF
   Serialized.One a -> OneF a
-  Serialized.Give c -> GiveF c
-  Serialized.And c c' -> AndF c c'
-  Serialized.Or c c' -> OrF c c'
-  Serialized.Cond k c c' -> CondF k c c'
-  Serialized.Scale k c -> ScaleF k c
-  Serialized.When p c -> WhenF p c
+  Serialized.Give c -> GiveF . Pure $ c
+  Serialized.And (c :: c' :: cs) -> DA.Foldable.foldr (\c f -> AndF (Pure c) (Impure f)) (AndF (Pure c) (Pure c')) cs
+  Serialized.And oneOrNil -> error "deserialize: Malformed `And` (should have at least two elements)"
+  Serialized.Or c c' -> OrF (Pure c) (Pure c')
+  Serialized.Cond k c c' -> CondF k (Pure c) (Pure c')
+  Serialized.Scale k c -> ScaleF k (Pure c)
+  Serialized.When p c -> WhenF p (Pure c)
 --  Serialized.Anytime p c -> AnytimeF p c
 --  Serialized.Until p c -> UntilF p c
 
 -- | Converts a `Claim` into a `Serializable.Claim`, so it can be written to the ledger.
 serialize : Claim Observation t a -> Serialized.Claim t a
-serialize = ana \case
-  Zero -> Serialized.ZeroF
-  One a -> Serialized.OneF a
-  Give c -> Serialized.GiveF c
-  And c c' -> Serialized.AndF c c'
-  Or c c' -> Serialized.OrF c c'
-  Cond k c c' -> Serialized.CondF k c c'
-  Scale k c -> Serialized.ScaleF k c
-  When p c -> Serialized.WhenF p c
+serialize = cata \case
```

Comment:
As above, why the change from `ana` to `cata` ?

Reply:
Both work, but it's more than anything for symmetry. If `deserialize` is an unfold (`ana`morphism), then `serialize` should be the dual, a fold (`cata`morphism). If you do this, you can fuse the two operations using a `hylo` (which combines `ana` followed by a `cata` - see type signature). You can still fuse passes if they're both folds, but it becomes trickier if you want to do other things in between, which is a typical pattern i.e. deserialize -> transform -> serialize.

Reply:
And funnily enough, I've now had to change this to `histo` (which is the dual of `futu` - not `cata`) for it to work.